File: GameAPI/src/main/java/me/clutchy/games/Game.java
package me.clutchy.games;

public interface Game {

    void onLobbyStart();
    void onLobbyEnd();
    void onLobbyTick();

    void onGameStarting();
    void onGameStart();
    void onGameEnd();
    void onGameTick();

    int minimumPlayers();
    int maximumPlayers();

    RunMode runMode();

    public enum RunMode {
        AUTOMATIC,
        MANUAL
    }
}

File: GameAPI/src/main/kotlin/me/clutchy/games/Game.kt
package me.clutchy.games

interface Game {

    fun onLobbyStart()
    fun onLobbyEnd()
    fun onLobbyTick()
    fun onGameStarting()
    fun onGameStart()
    fun onGameEnd()
    fun onGameTick()
    fun minimumPlayers(): Int = 2
    fun maximumPlayers(): Int = 10
    fun runMode(): RunMode = RunMode.MANUAL

    enum class RunMode {
        AUTOMATIC, MANUAL
    }
}
